### ISSUE TITLE
PersistentSubscriptionSettings.fromRevision only accepts START and bigint

### ISSUE THREAD

**Issue by bjorm** [EventStore/EventStore-Client-NodeJS#152]
I already asked this at https://discuss.eventstore.com/t/persistentsubscriptionsettings-fromrevision-only-accepts-start-and-bigint/3206 but so far I have not gotten any answers and it is important to us.

Could you explain why PersistentSubscriptionSettings.fromRevision [1] cannot be set to END ? In most cases we would not want to consume all events from the beginning. Would using ReadRevision [2] not make more sense? or is it that way because it is safer to set it to an explicit revision if we don’t start from the beginning?

Judging from the implementation [3] it does not look like an oversight.

We will soon be migrating our existing ESDB to ESDB cloud using https://replicator.eventstore.org. The replicator does not copy the persistent subscriptions and rather than specifying the explicit revision we would like them to start at the end. Also the server still seems to understand `-1` for `startFrom` so I don't see why the client does not.

[1] https://github.com/EventStore/EventStore-Client-NodeJS/blob/master/src/utils/persistentSubscriptionSettings.ts#L17
[2] https://github.com/EventStore/EventStore-Client-NodeJS/blob/master/src/types/index.ts#L35
[3] https://github.com/EventStore/EventStore-Client-NodeJS/blob/master/src/persistentSubscription/createPersistentSubscription.ts#L54

**Comment by George-Payne** [comment]
Hi Björn,

Thank you for the though issue, it is indeed an oversight. 

The fix is here:
https://github.com/EventStore/EventStore-Client-NodeJS/pull/154

and I'll make a patch release once its made it through reviews / QA


If you need a fix before then, you can work around it by setting the revision as so:
```ts
persistentSubscriptionSettingsFromDefaults({
    fromRevision: BigInt("18446744073709551615"),
})
```

**Comment by bjorm** [comment]
That is great, thanks!

**Comment by George-Payne** [comment]
This is now available in [1.1.2](https://www.npmjs.com/package/@eventstore/db-client/v/1.1.2)
